app/code/Magento/Downloadable/Observer/UpdateLinkPurchasedObserver.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Downloadable\Observer;
+
+use Magento\Framework\Event\ObserverInterface;
+
+/**
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
+ */
+class UpdateLinkPurchasedObserver implements ObserverInterface
+{
+    /**
+     * Core store config
+     * @var \Magento\Framework\App\Config\ScopeConfigInterface
+     */
+    protected $_scopeConfig;
+
+    /**
+     * @var \Magento\Downloadable\Model\ResourceModel\Link\Purchased\CollectionFactory
+     */
+    protected $_purchasedFactory;
+
+    /**
+     * @var \Magento\Framework\DataObject\Copy
+     */
+    protected $_objectCopyService;
```

Comment:
Please, use `private` for properties and methods instead of `protected`

Reply:
Methods scope updated.